naacl2013 archive video ingestion

xinru1414 opened the pull request.

mjpost commented:
Thanks for this! It looks good to me. Can you put example usage at the top of the script? It seems that it expects a directory that looks like this?

```
videos/
    N13-1001.m4v
    N13-1002.m4v
    ...
```

is that correct? What is the invocation, then?

mjpost commented:
This doesn't seem quite right in the preview:

* It mixes up the button with the plain-text link. The link here should be just plain-text

<img width="950" alt="image" src="https://user-images.githubusercontent.com/455256/147408963-c7e38aca-b6b4-4220-af91-88ff944daa8a.png">

* There should be a video button in the upper right
* This seems to have broken the video button in the listing files. See for example [here](https://aclanthology.org/volumes/D18-1/) (where it works) and [on the preview](https://preview.aclanthology.org/naacl-2013-video-ingestion/volumes/D18-1/) (where it doesn't work)

<img width="801" alt="image" src="https://user-images.githubusercontent.com/455256/147409025-fab2ee99-1bed-4dc9-a45c-63aca77f1827.png">

mjpost commented on bin/anthology/papers.py:
```diff
@@ -78,6 +79,16 @@ def pdf(self):
+            ]
+        return []
+
     def _parse_revision_or_errata(self, tag):
```
I think in `master` branch videos must be handled as general attachments, but I don't remember offhand. This is a good direction, though; you just have to figure out why it broke the current videos.

mbollmann commented:
What exactly are the changes in `bin/anthology/*.py` trying to achieve?

mjpost commented:
The goal is to represent videos that are hosted locally. We'll need to do a little work here though since this isn't quite right. 

Treating videos as separately named from attachments might be a good idea but the decision is separate from the changes here.

mbollmann commented on bin/anthology/utils.py:
```diff
@@ -437,15 +437,11 @@ def parse_element(xml_element):
-                "type": element.get("tag", "video"),
-                "url": element.get("href"),
-            }
+            value = element.get("href")
```
This doesn't look right; value is a dict here for a reason.

xinru1414 commented:
tested on localhost: 
the video icons on the proceedings page
<img width="1446" alt="Screen Shot 2022-01-13 at 8 01 03 PM" src="https://user-images.githubusercontent.com/27436920/149449390-efb7f1ee-5dff-4d38-8977-a299c1cb5873.png">
the video icon on the paper page
<img width="1378" alt="Screen Shot 2022-01-13 at 8 00 54 PM" src="https://user-images.githubusercontent.com/27436920/149449454-11bfe660-30a6-4385-8985-1959a01b8621.png">

mjpost commented:
Fantastic! They’re showing up in the preview, too. I’ll do a thorough review soon.

Looking through the [N13-1 preview](https://preview.aclanthology.org/naacl-2013-video-ingestion/volumes/N13-1), there is a chunk of missing videos (24–70). Any idea what happened here?

xinru1414 commented:
@mjpost videos (24–70) don't exist in the archive

mjpost commented on bin/anthology/utils.py:
```diff
@@ -437,15 +437,11 @@ def parse_element(xml_element):
-                "type": element.get("tag", "video"),
-                "url": element.get("href"),
-            }
+            value = element.get("href")
```
I'm guessing we don't use these other values here?

mbollmann replied:
These are the expected values for type "attachment", but "video" is now separate. I wonder if we might have a use for additional fields in the future, though, but it probably wouldn't be too difficult to refactor again later.

xinru1414 replied:
@mjpost @mbollmann Yes, it makes more sense to refactor in the future (with little effort).

mjpost reviewed (approved):
LGTM. I like how you've split out videos as a special kind of attachment.

One question that comes to mind, though, is whether this handles multiple video attachments? As we continue ingesting these, we will have situations where we have both our local mirror of a video and a remotely-hosted version (say on slideslive). Perhaps we can handle that situation when we come to it.

It's a shame that we're missing so many videos from NAACL 2013...

mjpost commented:
Let's wait till Monday to merge this in case @mbollmann or anyone else wants to comment.

xinru1414 commented:
@mjpost SG

mbollmann commented on bin/anthology/utils.py:
```diff
@@ -437,15 +437,11 @@ def parse_element(xml_element):
             value = {"filename": element.text, "src": element.get("src")}
         elif tag == "video":
             # Treat videos the same way as other attachments
```
This comment is now no longer true, should probably remove it

mbollmann commented on bin/anthology/papers.py:
```diff
@@ -15,6 +15,7 @@
 # limitations under the License.
 
 from functools import cached_property
+import sys
```
? I don't see this used

mbollmann commented on data/xml/Q13.xml:
```diff
@@ -35,6 +35,7 @@
       <pages>13–24</pages>
       <url hash="84ba73ea">Q13-1002</url>
       <bibkey>pitler-etal-2013-finding</bibkey>
+      <video tag="video" href="Q13-1002.mp4"/>
```
I'm not sure why we keep this `tag="video"` around. I've done a grep on all our current XML files and the value for `tag` is never something else than `"video"`. Since we also don't do anything with it in the code, I think we should just get rid of it now. @mjpost?

For reference:
```
$ grep "<video" *.xml | sed 's/.*tag="\([[:alpha:]]*\)".*/\1/' | sort | uniq -c                                                                                                                                 
   4010 video
```

mjpost replied:
Yes, @xinru1414, let’s get rid of `tag=“video”`. We’ve had this discussion elsewhere before.

I also agree it might be good to use a dictionary for video values. For example, we might want to have a field that distinguishes remote from local links. We don’t need it now so we could just refactor later, but it might be easier now to preserve the dictionary structure while it’s fresh in your memory.

akoehn replied:
When getting rid of the tag, remember to update the schema and remove tags from the video element here: https://github.com/acl-org/acl-anthology/blob/3ef7ced577d9c95feaf2cecc94cd33d13398eea1/data/xml/schema.rnc#L72

xinru1414 replied:
@akoehn yep SG thanks for the reminder

mbollmann commented on hugo/layouts/papers/list-entry-author.html:
```diff
@@ -19,9 +19,12 @@
+    {{- range $paper.attachment -}}
+      <a class="badge badge-attachment align-middle mr-1" href="{{ .url }}" data-toggle="tooltip" data-placement="top" title="{{ .type | humanize }}">{{ partial "attachment_repr.html" . }}</a>
+    {{- end -}}
+    {{ range $paper.video }}
```
Add the dashes here too for consistency (`{{- ... -}}`)

mbollmann commented on hugo/layouts/papers/list-entry.html:
```diff
@@ -20,9 +20,12 @@
+    {{- range $paper.attachment -}}
+      <a class="badge badge-attachment align-middle mr-1" href="{{ .url }}" data-toggle="tooltip" data-placement="top" title="{{ .type | humanize }}">{{ partial "attachment_repr.html" . }}</a>
+    {{- end -}}
+    {{ range $paper.video }}
```
Add the dashes here too for consistency (`{{- ... -}}`)

xinru1414 commented:
@mjpost @mbollmann @akoehn I believe I've addressed all comments. Specifically for removing `tag=“video”`, I've removed it from the schema, all python scripts and all xml files (If I missed anything plz lmk).

mbollmann reviewed (approved):
LGTM now. Very nice!

nschneid commented:
Very glad to see these videos restored!

Do you want feedback on broken links? E.g. the videos at https://preview.aclanthology.org/naacl-2013-video-ingestion/volumes/N13-4/ are giving 404s.

mjpost commented:
@nschneid good catch. These files are split into parts and the htaccess regex didn't catch it. I just made an update attempt. This should work now once the preview is pushed out.

xinru1414 commented:
@mjpost thanks for the regex fix!

mjpost commented:
It still wasn't quite working, but this latest update should do it.

xinru1414 commented:
@mjpost i think it's working now. i spot checked several links: https://preview.aclanthology.org/naacl-2013-video-ingestion/N13-4004.3.mp4 , https://preview.aclanthology.org/naacl-2013-video-ingestion/N13-4001.1.mp4 , https://preview.aclanthology.org/naacl-2013-video-ingestion/N13-4006.1.mp4

Shall I merge?

Also, after merging I think we can move on to archiving/processing more videos?

mjpost commented:
Sure—please make sure to trim the commit message to just a summary of the relevant commits.

And yes, let’s move on to other conferences. Can you make a spreadsheet to track the status? I suggest (conference x year).
